File: SportHub_server/SportHub.Core/Entities/AdminPost.cs
using SportHub.Core.Entities.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace SportHub.Core.Entities
{
    public class AdminPost : Entity
    {
        public string Title { get; set; }
        public string Text { get; set; }
        public int Duration { get; set; }
        public int Complexity { get; set; }
        public string ImageUrl { get; set; }
        public DateTime DateCreated { get; set; }
        public bool IsUpdated { get; set; }

        public int CategoryId { get; set; }
        public PostCategory PostCategory { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; }

        public AdminPost(int categoryId, string title, string text, int duration, int complexity,
            string imageUrl, DateTime dateCreated, bool isUpdated, Guid userId)
        {
            CategoryId = categoryId;
            Title = title;
            Text = text;
            Duration = duration;
            Complexity = complexity;
            ImageUrl = imageUrl;
            DateCreated = dateCreated;
            IsUpdated = isUpdated;
            UserId = userId;
        }
    }
}

File: SportHub_client/lib/entities/admin_post.dart
import 'package:SportHub_client/entities/user.dart';

class AdminPost {
  int id;
  String title;
  String text;
  int duration;
  int complexity;
  String imageUrl;
  String dateCreated;
  bool isUpdated;
  int categoryId;
  String userId;
  User user;

  AdminPost(
      {this.id,
      this.title,
      this.text,
      this.duration,
      this.complexity,
      this.imageUrl,
      this.dateCreated,
      this.isUpdated,
      this.categoryId,
      this.user,
      this.userId});

  AdminPost.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    title = json['title'];
    text = json['text'];
    duration = json['duration'];
    complexity = json['complexity'];
    imageUrl = json['imageUrl'];
    dateCreated = json['dateCreated'];
    isUpdated = json['isUpdated'];
    categoryId = json['categoryId'];
    user = json['user'] != null ? new User.fromJson(json['user']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['title'] = this.title;
    data['text'] = this.text;
    data['duration'] = this.duration;
    data['complexity'] = this.complexity;
    data['imageUrl'] = this.imageUrl;
    data['dateCreated'] = this.dateCreated;
    data['isUpdated'] = this.isUpdated;
    data['categoryId'] = this.categoryId;
    data['userId'] = this.userId;
    if (this.user != null) {
      data['user'] = this.user.toJson();
    }
    return data;
  }
}
